tests/pytests/test_compositejk.py

```diff
@@ -66,112 +101,196 @@ def test_composite_call(j_algo, k_algo, mols, request):
+                       "ref_iter": {
+                           "base": 8,
+                           "df_scf_guess": 11,   
+                           "scf_cosx_guess": 13, # assumes COSX_MAXITER_FINAL = -1   
+                       },
+                     }, id="DF-DirJ+LinK"),
+        pytest.param({ "options": { 
+                           "scf_type": "DFDIRJ+COSX",
+                           "screening": "schwarz",
+                           "maxiter": 30,
+                       },
+                       "ref_iter": {
+                           "base": 8, # for single initial COSX grid, i.e., COSX_MAXITER_FINAL = 0
+                           "scf_cosx_guess": 20, # for full convergence on both COSX grids, i.e., COSX_MAXITER_FINAL = -1   
+                       },
+                     }, id="DF-DirJ+COSX (MAXITER=30)"),
+        pytest.param({ "options": { 
+                           "scf_type": "DFDIRJ+COSX",
+                           "screening": "schwarz",
+                           "maxiter": 10,
+                       },
+                       "ref_iter": {
+                           "base": 8, # for single initial COSX grid, i.e., COSX_MAXITER_FINAL = 0
+                           "scf_cosx_guess": 20, # for full convergence on both COSX grids, i.e., COSX_MAXITER_FINAL = -1   
+                       },
+                     }, id="DF-DirJ+COSX (MAXITER=10)"),
+ 
+    ], 
 )
-def test_dfjcosk(inp, mols, request):
+@pytest.mark.parametrize("cosx_maxiter_final", [ -1, 0, 1 ])
+@pytest.mark.parametrize("scf_cosx_guess", [ True, False ])
+@pytest.mark.parametrize("df_scf_guess", [ True, False ])
+def test_cosx_maxiter_final(inp, opts, cosx_maxiter_final, scf_cosx_guess, df_scf_guess, tests, mols, request):
+    """Test the COSX_MAXITER_FINAL keyword in various situations via SCF calculations"""
+
+    test_id = request.node.callspec.id
+    
+    # basic settings configuration
+    molecule = mols[tests[inp]["molecule"]]
+    psi4.set_options({
+        "basis": "cc-pvdz",
+        "cosx_maxiter_final": cosx_maxiter_final, 
+        "df_scf_guess": df_scf_guess, 
+        "scf_cosx_guess": scf_cosx_guess, 
+        "save_jk": True
+    })
+    psi4.set_options(tests[inp]["options"])
+    psi4.set_options(opts["options"])
+
+    # DFDIRJ+COSX doesn't support either DF_SCF_GUESS or SCF_COSX_GUESS, so skip these
+    if opts["options"]["scf_type"] == "DFDIRJ+COSX" and (scf_cosx_guess or df_scf_guess): 
+        pytest.skip(f'{test_id} skipped: DFDIRJ+COSX doesnt support JK guesses') 
+
+    # if both guesses are set to True, run should throw exception... 
+    elif scf_cosx_guess and df_scf_guess:
+        with pytest.raises(Exception) as e_info:
+            E = psi4.energy(tests[inp]["method"], molecule=molecule)
+
+        # we keep this line just for printout purposes; should always pass if done correctly
+        assert compare(type(e_info), pytest.ExceptionInfo, f'{test_id} throws exception (both guesses are True)')
```

Comment:
Good idea, thanks! I just updated this for assertions which only involve the type `Exception`. Other tests use more specific exception types (e.g., `psi4.ValidationError`) and I left those alone for now, though I'm happy to update those as well if you believe it would be beneficial.